torch/testing/_internal/common_methods_invocations.py

```diff
@@ -7811,6 +7811,80 @@ def sample_inputs_diagonal_diag_embed(op_info, device, dtype, requires_grad, **k
     for shape, kwarg in chain(product(shapes_2d, kwargs_2d), product(shapes_3d, kwargs_3d)):
         yield SampleInput(make_arg(shape), kwargs=kwarg)
 
+def reference_inputs_diagonal(op_info, device, dtype, requires_grad, **kwargs):
```

Comment:
it seems like these inputs mostly repeat what sample_inputs_diagonal_diag_embed are already testing, reference inputs should be testing more interesting cases, not repeat existing coverage.

Reply:
all these are different compared to what sample inputs test for:

```py
    kwargs2d = (
        # dim1 > dim2 is allowed
        dict(dim1=1, dim2=0),
        # negative dims are allowed
        dict(dim1=-2, dim2=-1),
        # out of bounds offset should return an empty tensor in diagonal and
        # offset the diagonal in diag_embed
        dict(offset=100),
    )

    kwargs3d = kwargs2d + (
        # make sure we can index dims other than 0 and 1
        dict(offset=-1, dim1=0, dim2=2),
    )
```

the last comment is misleading tho, it should say something like "test non-sequential dims" (i'll fix this).

the most important ones are negative indexes and the large offset. the rest here test that there's no "dependency" between dims in the code